### PR TITLE
Move ExampleSpear and ExampleSpearProjectile out of Old directory

### PR DESCRIPTION
**Pull request by Naginif**
Referenced issue: #1403 

### What are the changes to ExampleMod?

Port ExampleSpear and ExampleSpearProjectile out of `Old/` directory. This is my first contribution to tModLoader so please let me know if there are any conventions that I missed or something I need to do different.

### Do these changes need additional documentation?
No

### CODE REVIEW

**Comment by cozykoko** [comment]
Isnt examplespear already in a pr #1097 ?

**Comment by Naginif** [comment]
@cozykoko dang I thought I checked for this in existing PRs but looks like I missed this.  Though that PR has been idle for 3 months... 

Should I close this then?

**Comment by JavidPack** [comment]
No, we'll look at them both and decide later, when we are looking into these PRs

**Comment by Naginif** [comment]
@JavidPack okay thank you 👍 I also went ahead and added most of the requested changes from #1097 that were relevant / that I understood

**Comment by Mirsario** [comment]
> 
> 
> @JavidPack okay thank you 👍 I also went ahead and added most of the requested changes from #1097 that were relevant / that I understood

My main request for that PR was to make the spear projectile's movement not be velocity-based, but instead be done through setting the position to an offset calculated through linear interpolation.

Like seen here:
https://github.com/Mirsario/TerrariaOverhaul/blob/1.4/Content/Projectiles/SpearProjectileBase.cs?ts=4

File: ExampleMod/Content/Projectiles/ExampleSpearProjectile.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+			// These dusts are added later, for the 'ExampleMod' effect
+			if (Main.rand.NextBool(3)) {
+				Dust dust = Dust.NewDustDirect(projectile.position, projectile.width, projectile.height, DustType<Sparkle>(),
+					projectile.velocity.X * 2f, projectile.velocity.Y * 2f, 200, Scale: 1.2f);
+			}
+			if (Main.rand.NextBool(4)) {
+				Dust dust = Dust.NewDustDirect(projectile.position, projectile.width, projectile.height, DustType<Sparkle>(),
+					0, 0, 254, Scale: 0.3f);
+			}
```
Comment (Mirsario) [inline ExampleMod/Content/Projectiles/ExampleSpearProjectile.cs:70]:
```suggestion
			// Avoid spawning dusts on dedicated servers
			if (!Main.dedServ) {
				// These dusts are added later, for the 'ExampleMod' effect
				if (Main.rand.NextBool(3)) {
					Dust dust = Dust.NewDustDirect(projectile.position, projectile.width, projectile.height, DustType<Sparkle>(), projectile.velocity.X * 2f, projectile.velocity.Y * 2f, 200, Scale: 1.2f);
				}

				if (Main.rand.NextBool(4)) {
					Dust dust = Dust.NewDustDirect(projectile.position, projectile.width, projectile.height, DustType<Sparkle>(), 0, 0, 254, Scale: 0.3f);
				}
			}
```

**Review by Mirsario — CHANGES_REQUESTED** [review]
> 
> 
> I also went ahead and added most of the requested changes from #1097 that were relevant / that I understood

My main request for that PR was to make the spear projectile's movement not be velocity-based, but instead be done through setting the position to an offset calculated through linear interpolation.

Like seen here:
https://github.com/Mirsario/TerrariaOverhaul/blob/1.4/Content/Projectiles/SpearProjectileBase.cs?ts=4

Frankly, you can just take this code, replace abstract and virtual parts with constants, comment it, add back the existing dust code, and that's it.

**Comment by Naginif** [comment]
@Mirsario I believe I made the update to the SpearProjectile to be movement based.  Please let me know if there is anything else I can do 😄

**Review by Mirsario — APPROVED** [review]
The branch wasn't compiling, please test things in the future. I've fixed it and other things, improving the item as a whole. Thanks.

Also, for context, we're running low on time for reviews, so I mostly just review things I'm pinged for in the discord.

**Comment by Naginif** [comment]
@Mirsario thank you I really appreciate the updates and review! 

I was having trouble compiling and I couldn't get anyone to respond to my build issues in the discord. I'm a Mac/Java developer at work and am not used to developing on windows or .NET yet. I tried even just copying/pasting EM from the 1.4 branch into the Mod Sources directory and was getting this error many times over. It looks like it's failing to find any of the build-in namespaces? I'd appreciate some help debugging this or pointing me somewhere I can get help.  

Here's just a snippet of one of the error messages and a stack trace
```
Error: C:\Users\Daniel\Documents\My Games\Terraria\ModLoader\Mod Sources\ExampleMod\Common\ExamplePlayerDrawLayer.cs(4,14) : error CS0234: The type or namespace name 'Content' does not exist in the namespace 'ReLogic' (are you missing an assembly reference?)
   at Terraria.ModLoader.Core.ModCompile.CompileMod(BuildingMod mod, String outputPath, IDictionary`2 refs, Boolean xna) in tModLoader\Terraria.ModLoader.Core\ModCompile.cs:line 680
   at Terraria.ModLoader.Core.ModCompile.BuildModForPlatform(BuildingMod mod, Boolean xna) in tModLoader\Terraria.ModLoader.Core\ModCompile.cs:line 554
   at Terraria.ModLoader.Core.ModCompile.Build(BuildingMod mod) in tModLoader\Terraria.ModLoader.Core\ModCompile.cs:line 412
   at Terraria.ModLoader.Core.ModCompile.Build(String modFolder) in tModLoader\Terraria.ModLoader.Core\ModCompile.cs:line 369
   at Terraria.ModLoader.UI.UIBuildMod.<>c__DisplayClass5_0.<Build>b__0(ModCompile mc) in tModLoader\Terraria.ModLoader.UI\UIBuildMod.cs:line 31
   at Terraria.ModLoader.UI.UIBuildMod.BuildMod(Action`1 buildAction, Boolean reload) in tModLoader\Terraria.ModLoader.UI\UIBuildMod.cs:line 67

**Pull request merged**